### ISSUE TITLE
Matching a non-empty pattern

### ISSUE THREAD

**Issue by rix0rrr** [comby-tools/comby#357]
I feel rather silly posting this issue, because it seems so simple that I must be overlooking something. Nevertheless, looking at the [documentation here](https://comby.dev/docs/syntax-reference) it really seems impossible to do what I want, so here goes.

## The situation

I'm trying to rewrite jQuery handlers from a legacy syntax to a recommended syntax. Effectively, this is the transformation I'm trying to do:

```
$('#bla').submit(function() { ... });
    →
$('#bla').on('submit', function() { ... });
```

So far, so straightforward. The problem is that jQuery methods are heavily overloaded, and so `$('#bla').submit()` **also** exists, but should **not** be rewritten.

## The attempt

So I'm trying the following:

```
comby '$(:[a]).submit(:[b])' '$(:[a]).on(\'submit\', :[b])' *.ts
```

And that kinda works:

<img width="586" alt="image" src="https://user-images.githubusercontent.com/524162/218567632-c2ee9730-b0f4-4a02-8b4c-37fd62e1b35b.png">

Except it also catches the case I don't want to catch:

<img width="521" alt="image" src="https://user-images.githubusercontent.com/524162/218566612-ff0c5f69-1099-4723-be96-4cd28f17680f.png">

Oops!

## The problem

So, it seems that I want to specify a pattern `:[b]`, such that `b` may not be empty (but I do want to preserve the lazy/automatic matching that the `:[b]` syntax would do.

Looking at the [matching syntax table](https://comby.dev/docs/syntax-reference), none of the patterns achieve this:

* `:[[b]]` - only matches indentifiers, so will definitely not match anything of the form `function() { ... }`
* `:[b:e]` - sounds like it might do what I want it to do, but doesn't (apparently it doesn't like the `function () { }` syntax?)
* `:[b.]` - doesn't match whitespace

Sooo... I guess I'm looking for a non-empty variant of `:[x]` ?

**Comment by rvantonder** [comment]
> `:[b:e]` - sounds like it might do what I want it to do, but doesn't (apparently it doesn't like the `function () { }`

Yeah this gets close, but the limitation is it won't match across the whitespace between `function() { }`. If you are sure that the second argument is always of the form `function() {...}` this corresponds to two expressions like `:[f:e] :[braces:e]` where `f` matches `function()` and `braces` match the block in braces `{...}`.

> Sooo... I guess I'm looking for a non-empty variant of :[x] ?

The alternative here is to [use a rule ](https://comby.dev/docs/advanced-usage) to case out on non-empty matches of `:[x]`. Something like `-rule 'where :[x] != ""'` should work on the command line?
